swiftnav/amb_kf.pyx

```diff
@@ -14,54 +14,11 @@ from libc.string cimport memcpy, memcmp, memset
 from almanac cimport *
 from almanac_c cimport *
 from gpstime cimport *
+from linear_algebra import *
 from gpstime_c cimport *
 from single_diff_c cimport *
 from dgnss_management_c cimport *
 
-def udu(M):
-  n = M.shape[0]
-  cdef np.ndarray[np.double_t, ndim=2, mode="c"] M_ = \
-    np.array(M, dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=2, mode="c"] U = \
-    np.empty((n,n), dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=1, mode="c"] D = \
-    np.empty(n, dtype=np.double)
-  amb_kf_c.udu(n, <double *> &M_[0,0], <double *> &U[0,0], <double *> &D[0])
-  return UDU_decomposition(U, D)
-
-def reconstruct_udu(ud):
-  n = ud.D.shape[0]
-
-  cdef np.ndarray[np.double_t, ndim=2, mode="c"] U = \
-    np.array(ud.U, dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=1, mode="c"] D = \
-    np.array(ud.D, dtype=np.double)
-
-  cdef np.ndarray[np.double_t, ndim=2, mode="c"] M = \
-    np.empty((n,n), dtype=np.double)
-
-  amb_kf_c.reconstruct_udu(n, <double *> &U[0,0], <double *> &D[0], <double *> &M[0,0])
-  return M
-
-def update_scalar_measurement(h, R, U, D):
-  state_dim = h.shape[0]
-
-  cdef np.ndarray[np.double_t, ndim=1, mode="c"] h_ = \
-    np.array(h, dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=2, mode="c"] U_ = \
-    np.array(U, dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=1, mode="c"] D_ = \
-    np.array(D, dtype=np.double)
-  cdef np.ndarray[np.double_t, ndim=1, mode="c"] k = \
-    np.empty(state_dim, dtype=np.double)
-
-  amb_kf_c.update_scalar_measurement(state_dim,
-                                       <double *> &h_[0],
-                                       <double> R,
-                                       <double *> &U_[0,0],
-                                       <double *> &D_[0],
-                                       <double *> &k[0])
-  return UDU_decomposition(U_, D_), k
```

Comment:
Are all these functions no longer necessary? 

Should we consider deprecating rather than deleting them?